### PR TITLE
V2: Enables more authentication

### PR DESCRIPTION
**Pull request by gdavison**
Enables additional authentication configurations

### CODE REVIEW

**Comment by ewbankkit** [comment]
Does this depend on #71 to eliminate the `golangci-lint` errors?

File: aws_config.go [hunk 1]
```diff
@@ -79,7 +78,19 @@ func GetAwsConfig(ctx context.Context, c *Config) (aws.Config, error) {
 		config.WithEC2IMDSClientEnableState(imdsEnableState),
 		config.WithHTTPClient(httpClient),
 		config.WithAPIOptions(apiOptions),
+		// FIXME: This should only be set for retrieving Creds
+		config.WithRetryer(func() aws.Retryer {
+			return aws.NopRetryer{}
+		}),
```
Comment (ewbankkit) [inline aws_config.go:84]: Does this mean that no retries are attempted for any service?

Comment (gdavison) [reply aws_config.go:84]: For now, though it will be only when retrieving credentials. I'm currently doing some refactoring to resolve that

**Comment by gdavison** [comment]
> Does this depend on #71 to eliminate the `golangci-lint` errors?

Yes

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

**Pull request merged**